lib/ex_unit/lib/ex_unit/cli_formatter.ex

```diff
@@ -146,13 +146,23 @@ defmodule ExUnit.CLIFormatter do
         {:module_finished, %ExUnit.TestModule{state: {:failed, failures}} = test_module},
         config
       ) do
-    tests_length = length(test_module.tests)
+    # The failed tests have already contributed to the counter,
+    # so we should only add the successful tests to the count
+    successful_tests_length =
+      test_module.tests
+      |> Enum.filter(&is_nil(&1.state))
+      |> length()
+
+    config = %{
+      config
+      | failure_counter: config.failure_counter + successful_tests_length
+    }
```

Comment:
```suggestion
    config = update_in(config.failure_counter, & &1 + successful_tests_length)
```

You could also merge the two suggestions into one:

    config = update_in(config.failure_counter, fn counter ->
      counter + Enum.count(test_module.tests, &is_nil(&1.state))
    end)

But I am not sure it is an improvement.

Reply:
I prefer this version; fixed!